imod/mf6/pkgbase.py

```diff
@@ -8,9 +9,42 @@
 
 import imod
 from imod.mf6.interfaces.ipackagebase import IPackageBase
+from imod.typing.grid import GridDataset
 
 TRANSPORT_PACKAGES = ("adv", "dsp", "ssm", "mst", "ist", "src")
 EXCHANGE_PACKAGES = "gwfgwf"
+ARGS_TO_EXCLUDE = ["validate"]
+
+
+def merge_unstructured_dataset(variables_to_merge):
```

Comment:
I would move this to a separate class. You could add this to grid.py class in the typing folder. And have a typeddispatch for the xr and xu versions of merge. It will also make it easier to remove this functionality when xugrid supports merging